src/anonymousUserTracking/criteriaCompletionChecker.ts

```diff
@@ -357,18 +357,26 @@ class CriteriaCompletionChecker {
 
   private getValueFromNestedObject(eventData: any, field: string): any {
     const valueFromObj = this.getFieldValue(eventData, field);
-    if (typeof valueFromObj === 'object') {
-      return Object.keys(valueFromObj).map((key) =>
-        this.getValueFromNestedObject(valueFromObj, key)
-      );
+    if (typeof valueFromObj === 'object' && valueFromObj !== null) {
```

Comment:
Can we add additional test coverage for this method?

Reply:
While reviewing the scenario for writing the test case, I realised that this scenario will never occur because our portal doesn't accept values as objects when creating criteria. Therefore, I have removed that method, and everything is working fine for all the criteria.